subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultDependencyConstraint.java

```diff
@@ -52,6 +52,12 @@ public DefaultDependencyConstraint(String group, String name, String version) {
         this.versionConstraint = new DefaultMutableVersionConstraint(version);
     }
 
+    public static DefaultDependencyConstraint of(String group, String name, Action<? super MutableVersionConstraint> versionSpec) {
```

Comment:
unit test?

Reply:
I'm not fan of unit tests which would just basically copy the implementation...